lib/src/observable_value.dart

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+/// A single variable with a stream emitting the variable's changing values.
+///
+/// Contains a single [value] which can be read and written as any other
+/// variable.
+///
+/// Whenever the value is set, the new value is emitted on the
+/// [updates] stream.
+/// Further, any new listener on the [updates] stream is initially sent the
+/// current [value].
+///
+/// Consider using immutable types for [value] since listeners will not
+/// be notified if the internal state of [value] changes, only if
+/// the [value] variable is written to.
+///
+/// Example:
+/// ```dart
+/// class ToggleWidget {
+///   final ObservableValue<bool> _isVisible = ObservableValue(true);
+///
+///   // ...
+///
+///   Stream<bool> get visibilityChanges => _isVisible.updates;
+///
+///   void toggleVisibility() {
+///     _isVisible.value ^= true; // XOR with true toggles value.
+///   }
+/// }
+/// ```
+abstract class ObservableValue<T> {
+  /// The current value.
+  ///
+  /// The [updates] stream emits the current value when listened to,
+  /// and the new value when `value` is set to a new value.
+  abstract T value;
+
+  /// A stream emitting the current value of [value].
+  ///
+  /// New listeners will receive an event containing the current value
+  /// of [value] as soon as possible after listening.
+  /// Every time [value] changes, they will then receive an event
+  /// with the new value.
+  ///
+  /// This stream can be listened to multiple times.
+  /// It will never emit an error event.
+  /// _It's not a broadcast stream, because cancelling a subscription
+  /// on the stream, and then quickly listening again, may cause the
+  /// same event (the current value) to be seen twice._
+  abstract final Stream<T> updates;
```

Comment:
I suggest to rename this to `values` or `valueStream` to emphasize it is not only updates to the values after listening, but we will get the value that was set so far. The fact that we get updates should be clear from the `Stream` type

Reply:
That would be `values`. No need to put `Stream` on the name of something with type `Stream`.